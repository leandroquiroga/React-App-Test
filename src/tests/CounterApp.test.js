import React from 'react';
import { shallow } from 'enzyme';
import CounterAPP from '../CounterApp';


describe('Testing on componentes <CounterApp />', () => {
    const value = 0;
    let wapper = shallow(<CounterAPP value={value} />);                                                                                                                                                                                                                                                                     

    beforeEach(() => {
        wapper = shallow(<CounterAPP value={value} />);
    });

    test('This test must show the component correctly', () => {
        expect(wapper).toMatchSnapshot();
    });

    test('This test must show the defect value (0) ', () => {
        const valueDefect = wapper.find('h2').text().trim();
        expect(Number(valueDefect)).toBe(value);
    });

    test('This test must increment of the value (+1)', () => {
        wapper.find('button').at(0).simulate('click');
        const value = wapper.find('h2').text().trim();
        expect(value).toBe('1')
    });

    test('This test must decrement of the value (-1)', () => {
        wapper.find('button').at(1).simulate('click');
        const valueDefault = wapper.find('h2').text().trim();
        expect(valueDefault).toBe('-1');
    });

    test('This test must reset of value by defect', () => {
        const wapper = shallow(<CounterAPP value={0} />);

        wapper.find('button').at(0).simulate('click');
        wapper.find('button').at(2).simulate('click');

        const valueCurrently = wapper.find('h2').text().trim();

        expect(valueCurrently).toBe('0')
    })
});