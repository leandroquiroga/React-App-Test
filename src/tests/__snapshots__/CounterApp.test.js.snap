// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on componentes <CounterApp /> This test must show the component correctly 1`] = `
<Fragment>
  <h1>
    Counter APP
  </h1>
  <h2>
    0
  </h2>
  <button
    className="button"
    onClick={[Function]}
  >
    Increment value
  </button>
  <button
    className="button"
    onClick={[Function]}
  >
    Decrement value
  </button>
  <button
    className="button"
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
